src/utils/commandRunner.ts

```diff
@@ -59,12 +59,23 @@ export class CommandRunner {
         `${executable} ${args}`,
         mountPaths
       );
-      runEnv = undefined;
+      runEnv = {};
     }
+    // Ensure we can parse the output
+    if (runEnv === undefined) {
+      runEnv = process.env;
+    }
+    runEnv['ANSIBLE_FORCE_COLOR'] = '0';
 
-    const currentWorkingDirectory = workingDirectory
-      ? workingDirectory
-      : URI.parse(this.context.workspaceFolder.uri).path;
+    let currentWorkingDirectory = workingDirectory;
+    if (!workingDirectory && this.context?.workspaceFolder.uri) {
```

Comment:
And here, `workspaceFolder` is not optional according to the interface.

Reply:
@tomaciazek the root cause of all those ugly modifications to the source was that I was not able to get a valid configuration object. I guess that if find the right way to instantiate it i will be able to remove those and avoid touching the code.